### PR TITLE
FIX export of amazon-payment orders

### PR DESCRIPTION
**Pull request by jppeter**
wait for completed payment in case of amazon payment to prevent destroyed addresses

### CODE REVIEW

File: Components/Export/Continuous/Entity/PlentymarketsExportEntityOrder.php [hunk 1]
```diff
@@ -115,8 +115,12 @@ public function __destruct()
 	 */
 	public function export()
 	{
-		$this->exportCustomer();
-		$this->exportOrder();
+		// wait for completed payment in case of amazon payment to prevent destroyed addresses
+		$methodOfPayment = $this->getMethodOfPaymentId();
+		if ($methodOfPayment != MOP_AMAZON_PAYMENT || $this->Order->getPaymentStatus()->getId() == 12) {
```
Comment (joconja) [inline Components/Export/Continuous/Entity/PlentymarketsExportEntityOrder.php]: 12 hardcoded?

**Comment by jppeter** [comment]
Yes, I think this is okay in this case. 12 is the ID of the payment-status "Komplett bezahlt" which should exist in every shopware-environment. The status is configurable inside the amazon-payment-plugin, 12 is the default-value here and the other options do not make sense in my opinion. A safer way could be to fetch this config...

**Comment by joconja** [comment]
Passt du das noch an?

**Comment by jppeter** [comment]
Wir könnten die zugehörige Config aus dem Amazon-Payments-Plugin ziehen und dann die 12 als Fallback nutzen. Besser?

**Comment by ioanamoldovan** [comment]
Hi,
ja, bitte es aus Amazon Payments Plugin ziehen. 

Gruß
Ioana Moldovan

**Comment by jochenmanz** [comment]
Hi,

could you update your pull request to use the amazon payment plugin config instead of the hardcoded value? 

Best regards,
Jochen

**Comment by bestit-baumann** [comment]
Hi, it's possible that a shopowner set an other status on APA.
I have seen a case, where a shopowner sets "Bezahlt mit Amazon" as a custom status for 

We just add "Erneut Authorisieren" as a new state to s_core_states.

The state - set by custom is found at:
- s_core_config_elements » name = "statePaymentCaptureDone"
- s_core_config_values » element_id -> value
If you reference this value - it is save, that the order was payed and done.

I hope i could help ;)

**Pull request closed without merging**